src/Lykke.AlgoStore.MatchingEngineAdapter.Abstractions/Services/Listening/StreamWrapper.cs

```diff
@@ -0,0 +1,249 @@
+﻿using Common.Log;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Domain.Listening.Requests;
+using ProtoBuf;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Services.Listening
```

Comment:
This class is very similar to https://github.com/LykkeCity/MatchingEngineConnector/blob/3ad771fcbc3c98f2f9489ad39f93706851fcc44e/src/Lykke.MatchingEngine.Connector/Tools/MatchingEngineSerializer.cs
Consider reusing existed

Reply:
The StreamWrapper was made because we discussed that algo instances should have as limited as possible access to the matching engine (and information in general). It also allows us to change the communication protocol at any time and support accepting only the messages we allow.

Therefore, I do not believe we can reuse the matching engine serializer here.